### PR TITLE
DCR bundles for apps pages

### PR DESCRIPTION
**Pull request by OllysCoding**
Closes https://github.com/guardian/dotcom-rendering/issues/7130

## What does this change?

Creates a 'single file' apps bundle as part of the DCAR migration.

To allow DCR to render for apps, we need to output the client-side code as a single bundle. This is because the apps allow for offline reading - we want all the javascript to be cached on our readers phones/tablets, so that any page downloaded for offline reading can render & work as expected without the need for an internet connection (nb. some client side code will still need connections for making API requests, etc).

This PR creates an 'apps' bundle which uses the `LimitChunkCountPlugin` to force webpack to output just a single chunk (file). We also add `Buffer` to the build using the `ProvidePlugin` as this will be required for [bridget in the future](https://github.com/guardian/dotcom-rendering/milestone/74).

We also modify `assets.ts` to allow us to include the apps bundle into a page in the future (https://github.com/guardian/dotcom-rendering/issues/7058), this involved slightly modifying the interface to allow for multiple platforms

```ts
/* Before (only for web) */
const getScriptArrayFromFile = getScriptsFromManifest(
	shouldServeVariantBundle,
);

getScriptArrayFromFile('index.js') // output: ["index.modern.xxx.js", "index.legacy.xxx.js"]

/* After (supports apps & web) */
/* Web */
const getScriptArrayFromFile = getScriptsFromManifest({
	platform: 'web'
	shouldServeVariantBundle,
});

getScriptArrayFromFile('index.js') // output: ["index.modern.xxx.js", "index.legacy.xxx.js"]


/* Apps */
const getScriptArrayFromFile = getScriptsFromManifest({
	platform: 'apps'
	/* Trying to add 'shouldServeVariantBundle' would error here */ 
});

getScriptArrayFromFile('index.js') // output:  ["index.apps.xxx.js"]
```

### Should we support 'variant' builds in the app?

In short? Maybe, but I haven't added it yet.

Lots of the use of the 'variant' build in DCR is related to optimising for CWV, which will be less of a priority in the apps - where JS will almost always be cached, and search engines aren't ranking us on performance.

That said, if we were to make other changes to our build system, being able to test them and monitor sentry for errors from the app would be useful. 

Given that we likely won't be doing this anytime soon, I've left this functionality out for now, though it would be relatively trivial to add it in the future. Perhaps this just adds more complexity, so I'd be interested on feedback for this point.

### CODE REVIEW

File: dotcom-rendering/src/server/dev-server.ts [hunk 1]
```diff
@@ -39,6 +41,9 @@ export const devServer = (): Handler => {
 			case '/FrontJSON':
 				return handleFrontJson(req, res, next);
 			default: {
+				// Do not redirect assets urls
+				if (req.url.match(ASSETS_URL)) return next();
```
Comment (OllysCoding) [inline dotcom-rendering/src/server/dev-server.ts:45]: A little drive-by bugfix/improvement, trying to load a 404ing assets URL would try and redirect you to a front... which would then error because the json was invalid.

**Review by mxdvl — APPROVED** [review]
Having worked on these files before, this all looks good!

It might be that it’s now prime time for a new type: 

```ts
export type Bundle = 'legacy' | 'modern' | 'variant' | 'apps';
```

**Review by AshCorr — APPROVED** [review]
👍

**Review by JamieB-gu — APPROVED** [review]

**Review by georgeblahblah — APPROVED** [review]
Looks great!

File: dotcom-rendering/scripts/webpack/webpack.config.browser.js [hunk 2]
```diff
@@ -89,44 +90,74 @@ const getLoaders = (bundle) => {
 					},
 				},
 			];
+		case 'apps':
```
Comment (arelra) [inline dotcom-rendering/scripts/webpack/webpack.config.browser.js:93]: Is the term `browser` for this file getting slightly overloaded and congested if it also includes apps.

This doesn't have to be now but could the 'apps' flow be separated into its own `webpack.config.apps.js` ?

Comment (OllysCoding) [reply dotcom-rendering/scripts/webpack/webpack.config.browser.js:93]: Hmmm, I'll take a think about this, though I do have a PR open to rename browser to client ... https://github.com/guardian/dotcom-rendering/pull/7125
